src/clients/SpokePoolClient.ts

```diff
@@ -334,6 +336,24 @@ export class SpokePoolClient extends BaseAbstractClient {
     return fills?.find((fill) => validateFillForDeposit(fill, deposit));
   }
 
+  private isEvmRepaymentValid(fill: Fill, matchedDeposit: Deposit): boolean {
+    // Slow fills don't result in repayments so they're always valid.
+    if (isSlowFill(fill)) {
+      return true;
+    }
+    // Lite chain deposits force repayment on origin chain.
+    const repaymentChainId = matchedDeposit.fromLiteChain ? fill.originChainId : fill.repaymentChainId;
+    // Return undefined if the requested repayment chain ID is not recognized by the hub pool.
+    const possibleRepaymentChainIds = this.configStoreClient?.getEnabledChains();
+    if (isDefined(possibleRepaymentChainIds) && !possibleRepaymentChainIds.includes(repaymentChainId)) {
```

Comment:
You might be able to consolidate this a little bit by defaulting to an empty array, though I'm unclear what is intended to happen here during test (i.e. when no ConfigStoreClient is available, so `possibleRepaymentChainIds` will revert to `undefined`. It looks like we just assume repayment to any chain is OK in that case?

```suggestion
    const possibleRepaymentChainIds = this.configStoreClient?.getEnabledChains() ?? [];
    if (!possibleRepaymentChainIds.includes(repaymentChainId)) {
```